### ISSUE TITLE
Feature request: User friendly mode Nickserv authentication for individual servers

### ISSUE THREAD

**Issue by sloanonlinux** [kvirc/KVIrc#2069]
I'm one of those smart people who uses a password manager for my Nickserv passwords, and I use special characters, and now I have to find all the special characters and put a backslash on it. Also, I want nickserv authentication possible for individual servers.

![Screenshot](http://i.imgur.com/3VFwKtr.png)

_modedited_ adding screenshot

![Another screenshot](http://i.imgur.com/qFR0IP4.png)

**Comment by un1versal** [comment]
> I'm one of those smart people who uses a password manager for my Nickserv passwords, and I use special characters, and now I have to find all the special characters and put a backslash on it. Also, I want nickserv authentication possible for individual servers.

Turn command line-friendly mode on., Im not sure how using some 3rd party password manager works with KVIrc, my guess is it doesnt. see related issue https://github.com/kvirc/KVIrc/issues/789

edited removed screenshot

**Comment by sloanonlinux** [comment]
My password manager simply copies the password into the clipboard for several seconds and is therefore compatible with anything that accepts pasting.

**Comment by un1versal** [comment]
> My password manager simply copies the password into the clipboard for several seconds and is therefore compatible with anything that accepts pasting.

Yes like I said, then turn on command line friendly mode to avoid adding back slashes before special characters

see https://github.com/kvirc/KVIrc/wiki/FAQ#im-trying-to-execute-a-command-me-msg-say--but-the-text-message-gets-truncatedi-get-a-kvs-error-why

**Comment by un1versal** [comment]
> Also, I want nickserv authentication possible for individual servers.

this is already possible, it works at least and theres several ways to go about this depending

![capture](https://cloud.githubusercontent.com/assets/3521959/17159507/b5341c92-5395-11e6-88e8-702f46bbbaf2.PNG)

![capture1](https://cloud.githubusercontent.com/assets/3521959/17159506/b533071c-5395-11e6-9ff2-28e47db28080.PNG)

![capture2](https://cloud.githubusercontent.com/assets/3521959/17159505/b5328314-5395-11e6-9e0a-3511b5e242f3.PNG)

If this is not what youre looking for join the old as sin request here https://github.com/kvirc/KVIrc/issues/1144

@sloanonlinux please check that command line friendly mode fixes your issues, the other problem is a duplicate anywho.

**Comment by sloanonlinux** [comment]
> Yes like I said, then turn on command line friendly mode to avoid adding back slashes before special characters

![Another screenshot](http://i.imgur.com/qFR0IP4.png)

**Comment by un1versal** [comment]
Well thats clear now at least what you meant in first place, and only took 5 posts. added that screenshot to main post as the one there explains nothing really useful.

OK marking this as a bug and as for the other request nickserv auth on every server thats already possible.

its a mess for sure, profiles and identity are completely separate, I wont hold my breath on any getting fixed soon #1144 has been open since 2011

**Comment by Heufneutje** [comment]
Just out of curiosity, what's the use case behind having different NickServ passwords for individual servers? Networks pretty much only have a single set of services.

Also yes, SASL is pretty much the preferred way to authenticate on networks that support it.

**Comment by un1versal** [comment]
> Just out of curiosity, what's the use case behind having different NickServ passwords for individual servers? Networks pretty much only have a single set of services.

Not long ago you replied to a request that has that answer, https://github.com/kvirc/KVIrc/issues/2056 or did you forget already?

**Comment by Heufneutje** [comment]
> Not long ago you replied to a request that has that answer, #2056 or did you forget already?
1. My name is nowhere in that issue.
2. That issue has literally 0 to do with having different NickServ passwords for individual servers on a network.

**Comment by un1versal** [comment]
> My name is nowhere in that issue.
> That issue has literally 0 to do with having different NickServ passwords for individual servers on a network.

I meant https://github.com/kvirc/KVIrc/issues/2036

**Comment by Heufneutje** [comment]
You can have multiple accounts on one network yeah, but you can specify multiple NickServ rules per network, which include the nickname.

**Comment by un1versal** [comment]
I know you can, but you asked wht the usage scenario was, that is one,

**Comment by IceN9ne** [comment]
> User friendly mode for Identify Command box

This box is designed to be a command, not a password.  If we made it user-friendly-mode style, then you wouldn't be able to put useful commands into the box like `identify $my.network $my.nick` for people that created their own alias to pull the passwords from a file and not store the passwords in a kvirc config.  Also, allowing for easier changing of passwords/nicks.

Escaping the special characters (I think you only need to do $ and %) is a one time thing for most people and changing the behavior of this will completely remove a beneficial feature that some of us use due to a mild-hassle of typing a couple of extra characters.

> Also, I want nickserv authentication possible for individual servers.

As @Heufneutje said, the solution for this is to go to the 'Network Details and Configuration' and add additional NickServ rules to handle each nick on the network.  This makes more sense to me as NickServ is a network wide feature, not a server specific feature.

If we added it to the 'Server Details and Configuration', this would only prevent you from using the nick on other servers, not add any additional abilities.  If you relegate said nick to only being on one server, then you lose the ability to /nick change it on another server and have the auto-identify rule fire.

Did I miss anything else in this ticket?
